pythran/toolchain.py

```diff
@@ -53,6 +53,13 @@ def CCompiler_customize(self, _, need_cxx=0):
             self.compiler_so.remove('-Wstrict-prototypes')
         except (AttributeError, ValueError):
             pass
+    # Remove -arch i386 if 'x86_64' is specified.
+    try:
+        if 'x86_64' in ' '.join(self.compiler_so):
+            # Ugly shortcut to remove '-arch', 'i386' from the list of arguments:
```

Comment:
what about:

```
archs = defaultdict(list)
for i, flag in enumerate(self.compiler_so[1:]):
    if self.compiler_so[i] == '-arch':
        archs[flag].append(i)

if 'x86_64' in archs and 'i386' in archs:
    for i in archs['i386']:
        self.compiler_so[i] = 'x86_64'
```

or something similar?

Reply:
OK I rewrote it according to your example.